contracts/Content/UniqueContent.sol

```diff
@@ -1,65 +1,88 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import "@openzeppelin/contracts-upgradeable/utils/introspection/ERC165CheckerUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/introspection/ERC165StorageUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC1155/utils/ERC1155HolderUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/interfaces/IERC2981Upgradeable.sol";
-import "../libraries/LibAsset.sol";
-import "./MultipleRoyalties.sol";
 import "./interfaces/IMultipleRoyalties.sol";
 import "./interfaces/IUniqueContent.sol";
+import "./interfaces/IUniqueContentStorage.sol";
 import "./interfaces/IContent.sol";
+import "../libraries/LibRoyalty.sol";
 
-contract UniqueContent is IUniqueContent, IMultipleRoyalties, MultipleRoyalties, ERC721Upgradeable, ERC1155HolderUpgradeable, IERC2981Upgradeable {
+contract UniqueContent is IUniqueContent, IMultipleRoyalties, ERC721Upgradeable, ERC1155HolderUpgradeable, IERC2981Upgradeable, ERC165StorageUpgradeable {
 
     /***************** Stored Variables *****************/
-    mapping(uint256 => LibAsset.UniqueAsset) uniqueAssetInfo;
+    IUniqueContentStorage uniqueContentStorage;
     uint256 private uniqueIdsCounter;
+
+    using ERC165CheckerUpgradeable for address;
+
+    function initialize(
+        address _uniqueContentStorage,
```

Comment:
this needs to me initialize(name, symbol, contentStorage)

The ERC721 init expects (name, symbol) and so because we're extending it, the uniqueContent init therefore has to be (name, symbol, ... other new variables).